plugin/emacs/flycheck-quicklintjs.el

```diff
@@ -68,6 +68,7 @@
 https://quick-lint-js.com"
   :command ("quick-lint-js"
             "--output-format=emacs-lisp"
+            "--path-for-config-search" (eval buffer-file-name)
```

Comment:
maybe blocking: in emacs, a buffer is just utf8 encoded data in memory, and it's not necessarily tied to a file, thus buffer-file-name would return nil in such situation, however, this is a rare case, i personally don't know any workflow that would end up in such scenario, a way to reproducing such case:
`C-x b tmpbuf RET RET M-x js-mode`

my concerning with this is flycheck dont handling `nil` in this case, which would crash the plugin, or if it handles as an empty string, quick-lint-js fails would fail to run due `--path-for-config-search` without a value, right?

Reply:
Good point! (Vim has the same thing. I should see how we handle this case in Vim...)

Reply:
```suggestion
            (eval (if-let ((bfn (buffer-file-name))) `("--path-for-config-search" ,bfn) ""))
```
i bet this wouldnt work because we are already quoted by flycheck, but something in this lines